tools/objinfo.ml

```diff
@@ -266,12 +273,17 @@ let dump_obj filename =
     seek_in ic cu_pos;
     let cu = (input_value ic : compilation_unit) in
     close_in ic;
+    if cu.cu_flat_float_arrays && not Config.flat_float_array then begin
+
+    end;
```

Comment:
This strange conditional should be replaced with a call to `check_flat_float_arrays`.